### ISSUE TITLE
Display warning when apps have enabled filesystem=host and filesystem=home

### ISSUE THREAD

**Issue by digitalethics** [tchx84/Flatseal#163]
Security is only meaningful for less technically-inclined users when it is enabled _by default_. As I believe Flatseal seeks to fill that gap, however, by virtue of its design only _documents_ the Flatpak permissions model through a neat GUI, it could do better by _issuing a warning_ when Flatpaks come with security risks for end users (either through malicious actors or ignorance on the part of the developers or packagers). In this regard, it would be helpful to clearly indicate a warning in the Flatseal user interface if an app comes with **filesystem=host** and **filesystem=home** permissions enabled so that users can revoke those permissions immediately.

**Comment by tchx84** [comment]
In principle I agree, but I wonder whether is possible to achieve something that is non-invasive (e.g. taking an extreme example, it would be a horrible experience to go into Flatseal and get spammed with warning pop ups like it's windows explorer).

Even if we can think of something subtle, e.g. say a subtle color cue to highlight "sensitive" permissions, I wonder what the criteria would be to classify these permissions as such, given that the only 100% secure application is the one with no permissions at all (and 100% relying on portals instead).

Just thinking aloud...

**Comment by Arxcis** [comment]
Even more subtle than a subtle warning, could be a subtle tip.  

**Example:**
>Tip: Apps that support the [File Chooser Portal](https://flatpak.github.io/xdg-desktop-portal/portal-docs.html#gdbus-org.freedesktop.portal.FileChooser), does not need any filesystem permissions.

The idea of a tip is to encourage good behaviour, by presenting a better alternative, instead of using warnings, to scare people away from the bad behaviour.

**Comment by Arxcis** [comment]
@tchx84 Having said that, I would be totally fine if you declared this **out-of-scope** for Flatseal. It certainly sounds like a classic scope creep to me. I mean it is a big step to go from Flatseal having no oppinions, to having oppinions about these things.

I would rather have stats about which apps are good or bad, hosted on a site like https://www.protondb.com/, just for Flatpaks.

**Comment by tchx84** [comment]
> Even more subtle than a subtle warning, could be a subtle tip.
> 
> **Example:**
> 
> > Tip: Apps that support the [File Chooser Portal](https://flatpak.github.io/xdg-desktop-portal/portal-docs.html#gdbus-org.freedesktop.portal.FileChooser), does not need any filesystem permissions.
> 
> The idea of a tip is to encourage good behaviour, by presenting a better alternative, instead of using warnings, to scare people away from the bad behaviour.

The thing is, a tip like that, is not really actionable for a regular Flatseal user, as only the app developer can do something about it.

In a sense, Flatseal gives the responsibility (to decide whether it's worth compromising security by enabling something) back to the user. In an ideal world, users would understand this, but of course is not always the case. That's why I like, in spirit, the ideas presented here, but we just need to find a better way to convey this message.

**Comment by Arxcis** [comment]
@digitalethics By the way, your concern inspired me to create a tool to visualize how bad the problem of `--filesystem=home`-permissions are.

Turns out: Not so bad --> https://arxcis.github.io/flatstat/#filesystem.

It has imrpoved a lot during the years.

**Comment by tchx84** [comment]
> @digitalethics By the way, your concern inspired me to create a tool to visualize how bad the problem of `--filesystem=home`-permissions are.
> 
> Turns out: Not so bad --> https://arxcis.github.io/flatstat/#filesystem.
> 
> It has imrpoved a lot during the years.

very interesting data, you should share it on https://discourse.flathub.org/

**Comment by Arxcis** [comment]
@tchx84 I will share it. Thanks. 

Btw, I updated `flatstat` to show more than just graphs. Now you can discover the list of all apps with the `--filesystem=host` permission, sorted by Github star count --> https://arxcis.github.io/flatstat/search/?filter=filesystem-host

**Comment by stpnwf** [comment]
I think a compromise that could be within scope can be to add an Info icon next to each permission entry/category, where upon hover/click a brief ELI5 or ELI10 explanation (and if relevant, a Warning) is displayed with what each permission entry and/or category is about. This, I believe, is non intrusive and can still manage to be relevant/educational to less tech savvy users who are interested. 

I think this is relevant because, for example, it seems that a workaround for getting support for the system's default theme in some Flatpaks is to grant `filesystem=host` or `filesystem=home` permission to an application. Now, I personally do not know all the implications that granting this has, but I know enough to deduce this can be a potential security risk, so I only enable this to a few applications that I trust.
I do not know if other users of Flatseal in the same situation will show a similar judgement, or even if that judgement is prudent. Hence why an Info button with ELI5 information can be a good solution to give the user a taste of the implications his actions can have.

**Comment by nekohayo** [comment]
You could also maybe show a "⚠" character/emoji or icon in the listview of the apps, next to the app name, and show the details in a GtkTooltip when hovering it. A GtkInfobar could be shown inside (at the top) the app's details pane when the app is clicked.

**Comment by tchx84** [comment]
Having added those icons for displaying user and global overrides... I think I probably need to be more careful about adding more icons.

**Comment by rusty-snake** [comment]
Alternatively you could add an report function which opens a pop-up window that lists dangerous permissions together with an description.
